Firestore/core/test/firebase/firestore/remote/grpc_streaming_reader_test.cc

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <initializer_list>
+#include <memory>
+#include <vector>
+
+#include "Firestore/core/src/firebase/firestore/remote/grpc_unary_call.h"
+#include "Firestore/core/src/firebase/firestore/util/async_queue.h"
+#include "Firestore/core/test/firebase/firestore/util/grpc_stream_tester.h"
+#include "absl/types/optional.h"
+#include "grpcpp/support/byte_buffer.h"
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace remote {
+
+using util::AsyncQueue;
+using util::CompletionResult;
+using util::GrpcStreamTester;
+using util::Status;
+using util::CompletionResult::Error;
+using util::CompletionResult::Ok;
+
+class GrpcStreamingReaderTest : public testing::Test {
+ public:
+  GrpcStreamingReaderTest() : reader_{tester_.CreateStreamingReader()} {
+    reader_->Start([this](const Status& status,
+                          const std::vector<grpc::ByteBuffer>& responses) {
+      status_ = status;
+      responses_ = responses;
+    });
+  }
+
+  ~GrpcStreamingReaderTest() {
+    tester_.Shutdown();
+  }
+
+  GrpcStreamingReader& reader() {
+    return *reader_;
+  }
+
+  AsyncQueue& worker_queue() {
+    return tester_.worker_queue();
```

Comment:
In test fixtures, is normal to *not* make private things private. This would allow you to eliminate this method (and the others like it).

Here's an example of what I'm talking about: http://go/totw/122